app/models/game.rb

```diff
@@ -50,3 +50,17 @@ def populate_board
   end
 end
 
+def puts_in_check(target_x, target_y)
+  if user == game.white_id
+    game.black_id.pieces.each do |piece|
+      return true if piece.valid_move?(target_x, target_y)
+    end
+  else
+    game.white_id.pieces.each do |piece|
+      return true if piece.valid_move?(target_x, target_y)
+    end      
+  end
+  return false
+end
+
```

Comment:
I'm not sure this will work. This is checking the place that the piece moves to, but there are two cases that need to be accounted for. 

1. The place a user is trying to move the king
2. If moving a non-king piece opens up their king to attack. 

so this logic works for the first case, where the player is moving the king, but not for the second case where the move of a non-king opens the king to check. 

This code sort of mimics what I had in an earlier pull request for the initial check logic. when that logic is folded into valid move, it should return that the move isn't valid if the king is put in check actually, which is what your card asked you to do. ugh, that was bad planning on our part- I should have seen that when I was writing valid move. so the next task is to kind of say the opposite like... the opposing player has put your king in check. Sort of updating the state of the game, so the user is told at the beginning of the turn that they're in check. i think that might be what ash meant by hooking the check logic into the game controller.